notebooks/cugraph_benchmarks/release.ipynb

```diff
@@ -0,0 +1,628 @@
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Timing \n",
+    "What is not timed:  Reading the data</p>\n",
+    "What is timmed: (1) creating a Graph, (2) running the algorithm"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Import Modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# system and other\n",
+    "import gc\n",
+    "import os\n",
+    "import time\n",
+    "import numpy as np\n",
+    "\n",
+    "# rapids\n",
+    "import cugraph\n",
+    "import cudf\n",
+    "\n",
+    "# NetworkX libraries\n",
+    "import networkx as nx\n",
+    "\n",
+    "# MTX file reader\n",
+    "from scipy.io import mmread"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try: \n",
+    "    import community\n",
+    "except ModuleNotFoundError:\n",
+    "    os.system('pip install python-louvain')\n",
+    "    import community"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try: \n",
+    "    import matplotlib\n",
+    "except ModuleNotFoundError:\n",
+    "    os.system('pip install matplotlib')\n",
+    "\n",
+    "import matplotlib.pyplot as plt; plt.rcdefaults()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define the test data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Test File\n",
+    "data = {\n",
+    "    'preferentialAttachment' : './data/preferentialAttachment.mtx',\n",
+    "    'dblp'                   : './data/dblp-2010.mtx',\n",
+    "    'coPapersCiteseer'       : './data/coPapersCiteseer.mtx',\n",
+    "    'as-Skitter'             : './data/as-Skitter.mtx'\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Read data\n",
+    "The data is read in once once and used for both cuGraph and NetworkX."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Data reader - the file format is MTX, so we will use the reader from SciPy\n",
+    "def read_data(datafile):\n",
+    "    print('Reading ' + str(datafile) + '...')\n",
+    "    M = mmread(datafile).asfptype()\n",
+    "\n",
+    "    _gdf = cudf.DataFrame()\n",
+    "    _gdf['src'] = M.row\n",
+    "    _gdf['dst'] = M.col\n",
+    "    \n",
+    "    return _gdf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Graph functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# NetworkX\n",
+    "def create_nx_graph(_df):\n",
```

Comment:
it would be.  It was a quick retro fit based on Nx needing DiGraph.   
Updated and added text:

There are two types of graphs created:
Directed Graphs - calls to create_xx_digraph
Undirected Graphs - calls to create_xx_ugraph <- fully syemmeterized